lib/voorhees/json_api.ex

```diff
@@ -55,17 +55,41 @@ defmodule Voorhees.JSONApi do
   defp error_message(:ok), do: ""
   defp error_message({:error, message}), do: "Payload did not match expected\n\n" <> message
 
+  defp compare_property(actual, expected, property_name, options) do
+    actual_value = Map.fetch(actual, property_name)
+    expected_value =  Map.fetch(expected, property_name)
+
+    case {actual_value, expected_value} do
+      {:error, :error} -> :ok
+      {:error, _} -> compare_resources(actual_value, expected_value, options)
+      {_, :error} -> nil
+      {_, _} ->
+         compare_resources(elem(actual_value, 1), elem(expected_value, 1), options)
+         |> case do
+           {:error, message} ->
+             {:error, "\"#{property_name}\" did not match expected\n" <> message}
+           :ok -> :ok
+           {:ok, _} -> :ok # compare_resources_list returns this tuple due to reduce function
+         end
+    end
+  end
+
+  defp merge_results(existing, new) do
+    case {existing, new} do
+      {:ok, _} -> new
+      {_, :ok} -> existing
+      _ ->
+        {:error, existing_message} = existing
+        {:error, new_message} = new
+        {:error, elem(existing, 1) <> elem(new, 1)}
+    end
```

Comment:
I rather use pattern matching than elem. I would even add the case right in `compare_payloads`, as it is not going to be anywhere else but there, saving a hop

```
case {existing, new} do
  {:ok, state} -> state
  {state, :ok} -> state
  {{:error, existing_message}, {:error, new_message}} ->
    {:error, existing_message <> new_message}
end
```